tools/compatgen/main.go

```diff
@@ -12,11 +12,12 @@ import (
 
 func main() {
 	var output string
+	var minCudaVersion string
 
 	var rootCmd = &cobra.Command{
 		Use:   "compatgen {cuda|torch|tensorflow}",
 		Short: "Generate compatibility matrix for Cog base images",
-		Args:  cobra.ExactArgs(1),
+		Args:  cobra.MinimumNArgs(1),
```

Comment:
The way Cobra handles CLIs, flags aren't considered arguments. They're parsed out and captured before the remaining args are passed. 


```suggestion
		Args:  cobra.ExactArgs(1),
```